python/katana/example_data.py

```diff
@@ -66,3 +68,41 @@ def get_input_as_url(rel_path) -> str:
     """
     path = get_input(rel_path).resolve()
     return f"file://{path}"
+
+
+def parse_version(source) -> dict:
+    s = re.split(r"_", source[4:])
+    version = {}
+    version["Numbers"] = []
+    version["Branches"] = []
+    if len(s) == 0:
+        version["Numbers"] = [0]
+        version["Branches"] = ["."]
+        return version
+    i = 0
+    while i < len(s):
+        version["Numbers"].append(int(s[i]))
+        i += 1
+        if i < len(s):
+            version["Branches"].append(s[i])
+            i += 1
+        else:
+            version["Branches"].append(".")
+    return version
```

Comment:
I tried namedtuple, which is immutable. dict fits better here, IMO.

Reply:
Why do you need mutability? I really think this kind of version information should be handled functionally instead of using mutable. Also note that just because the namedtuple is immutable, doesn't mean that a list stored in it is immutable. So you can still append elements like you are here. But you would need to restructure slightly for a few things.

Reply:
I think we have a different emphasis. Version is used as a variable to grow while parsing the string. Strictly, namedtuple can still be mutated by replacing its attributes. I felt dict is already sufficient for the simple purpose of parsing versions and passing down to the request. It can be changed to namedtuple if you insist.